hive-metastore/src/main/java/org/apache/iceberg/hive/HiveTableOperations.java

```diff
@@ -203,8 +218,23 @@ protected void doCommit(TableMetadata base, TableMetadata metadata) {
           .orElseGet(ImmutableMap::of);
       setHmsTableParameters(newMetadataLocation, tbl, metadata.properties(), removedProps, hiveEngineEnabled, summary);
 
-      persistTable(tbl, updateHiveTable);
-      threw = false;
+      try {
+        persistTable(tbl, updateHiveTable);
+        commitStatus = CommitStatus.SUCCESS;
+      } catch (Throwable persistFailure) {
+        LOG.error("Cannot tell if commit to {}.{} succeeded, attempting to reconnect and check.",
+            database, tableName, persistFailure);
+        commitStatus = checkCommitStatus(newMetadataLocation, metadata);
+        switch (commitStatus) {
+          case SUCCESS:
+            return;
```

Comment:
It seems strange to return directly from here instead of breaking. The outer finally block uses the commit status and should be called regardless so I think this just looks more confusing than is needed.

Reply:
I just thought it had more symmetry with the other cases, since they are all function exits explicitly as well. I'll switch it to a break though if you like that better.